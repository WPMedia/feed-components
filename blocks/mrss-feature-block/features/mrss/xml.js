import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import get from 'lodash/get'
import moment from 'moment'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { BuildContent } from '@wpmedia/feeds-content-elements'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'
import { buildResizerURL } from '@wpmedia/feeds-resizer'
import { findVideo } from '@wpmedia/feeds-find-video-stream'

const jmespath = require('jmespath')

const rssTemplate = (
  elements,
  {
    channelTitle,
    channelDescription,
    channelPath,
    channelCopyright,
    channelTTL,
    channelUpdatePeriod,
    channelUpdateFrequency,
    channelCategory,
    channelLogo,
    imageTitle,
    imageCaption,
    imageCredits,
    itemTitle,
    itemDescription,
    pubDate,
    itemCredits,
    itemCategory,
    includeContent,
    resizerURL,
    domain,
    feedTitle,
    feedLanguage,
    videoInfo,
    mrssBuildContent,
  },
) => ({
  rss: {
    '@xmlns:atom': 'http://www.w3.org/2005/Atom',
    '@version': '2.0',
    '@xmlns:media': 'http://search.yahoo.com/mrss/',
    channel: {
      title: `${channelTitle || feedTitle + ' Videos'}`,
      link: domain,
      'atom:link': {
        '@href': `${domain}${channelPath}`,
        '@rel': 'self',
        '@type': 'application/rss+xml',
      },
      description: channelDescription || feedTitle,
      pubDate: moment.utc(Date.now()).format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
      ...(channelLogo && {
        image: {
          url: buildResizerURL(channelLogo, resizerKey, resizerURL),
          title: `${channelTitle || feedTitle + ' Videos'}`,
          link: domain,
        },
      }),

      item: elements.map((s) => {
        const url = `${domain}${s.website_url || s.canonical_url}`
        const img =
          s.promo_items && (s.promo_items.basic || s.promo_items.lead_art)
        const videoStream = findVideo(s, videoInfo)

        return {
          title: `${jmespath.search(s, itemTitle)}`,
          link: url,
          ...(itemCredits &&
            (jmespath.search(img, itemCredits) || []).length && {
              'media:credit': {
                '@role': 'producer',
                $: jmespath.search(img, itemCredits).join(','),
              },
            }),
          ...(itemDescription && {
            description: jmespath.search(s, itemDescription),
          }),
          guid: {
            '@isPermaLink': false,
            '#': url,
          },
          referenceid: s._id,
          pubDate: moment
            .utc(s[pubDate])
            .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),

          'media:content': {
            '@': {
              isDefault: 'true',
              ...(s.duration && {
                duration: Math.trunc(s.duration / 1000),
              }),
              ...(videoStream && {
                ...(videoInfo.url && { url: videoInfo.url }),
                ...(videoInfo.height && { height: videoInfo.height }),
                ...(videoInfo.width && { width: videoInfo.width }),
                ...(videoInfo.bitrate && { bitrate: videoInfo.bitrate }),
                ...(videoInfo.stream_type && {
                  type: videoInfo.stream_type,
                  '#': videoStream,
                }),
              }),
            },
          },
          'media:keywords': (
            jmespath.search(s, 'taxonomy.seo_keywords[*]') || []
          ).join(','),
          ...(s.description ||
            (s.subheadlines &&
              s.subheadlines.basic && {
                'media:caption': {
                  $: s.description || s.subheadlines.basic,
                },
              })),
          'media:transcript': s.transcript,
          ...(s.taxonomy &&
            s.taxonomy.primary_section &&
            s.taxonomy.primary_section.name && {
              'media:category': s.taxonomy.primary_section.name,
            }),
          ...(img &&
            img.url && {
              'media:thumbnail': {
                '@url': buildResizerURL(img.url, resizerKey, resizerURL),
              },
            }),
        }
      }),
    },
  },
})

export function Mrss({ globalContent, customFields, arcSite }) {
  const {
    resizerURL = '',
    feedDomainURL = '',
    feedTitle = '',
    feedLanguage = '',
  } = getProperties(arcSite)

  function MrssBuildContent() {
    BuildContent.call(this)
  }
  const mrssBuildContent = new MrssBuildContent()

  // can't return null for xml return type, must return valid xml template
  return rssTemplate(get(globalContent, 'content_elements', []), {
    ...customFields,
    resizerURL,
    domain: feedDomainURL,
    feedTitle,
    feedLanguage,
    mrssBuildContent,
  })
}

Mrss.propTypes = {
  customFields: PropTypes.shape({
    channelPath: PropTypes.string.tag({
      label: 'Path',
      group: 'Channel',
      description:
        'Path to the feed, excluding the domain, defaults to /arcio/mrss',
      defaultValue: '/arcio/mrss/',
    }),
    selectVideo: PropTypes.kvp.tag({
      label: 'Select video using',
      group: 'Video',
      description:
        'This criteria is used to filter videos encoded in the streams array',
      defaultValue: {
        height: '',
        width: '',
        bitrate: 5400,
        stream_type: 'mp4',
      },
    }),
    ...generatePropsForFeed('rss', PropTypes),
  }),
}
Mrss.label = 'MRSS'
export default Consumer(Mrss)
