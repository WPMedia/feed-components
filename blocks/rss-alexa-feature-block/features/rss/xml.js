import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import get from 'lodash/get'
import moment from 'moment'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { BuildContent } from '@wpmedia/feeds-content-elements'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'
import { buildResizerURL } from '@wpmedia/feeds-resizer'

const jmespath = require('jmespath')

const cheerio = require('cheerio')

const getURLType = (url) => {
  const imgRegex = new RegExp(/(jpe?g|png|webp|PNG|JPG|WEBP)$/);
  const audioRegex = new RegExp('^(https?|ftp|file):\/\/(www.)?(.*?)\.(mp3)$');
  const videoRegex = new RegExp('^(https?|ftp|file):\/\/(www.)?(.*?)\.(mp4|ts)$');
  const uri = url.split(".");
  const type = uri[uri.length -1];
  const defaultType =  {label: 'Audio',type: `audio/${type}`};
  if(imgRegex.test(url)) {
    return {label: 'image',type: `image/${type}`};
  }
  if(audioRegex.test(url)){
    return {label: 'Audio',type: `audio/${type}`};
  }
  if(videoRegex.test(url)){
    return {label: 'Video',type: `video/${type}`};
  }
  
  return defaultType;
}

const rssTemplate = (
  elements,
  {
    audioAvailable,
    channelTitle,
    channelDescription,
    channelCopyright,
    channelTTL,
    channelCategory,
    channelLogo,
    itemTitle,
    pubDate,
    itemCategory,
    includeContent,
    resizerURL,
    resizerWidth,
    resizerHeight,
    domain,
    feedTitle,
    feedLanguage,
    rssBuildContent,
  },
) => ({
  rss: {
    '@version': '2.0',
    channel: {
      ...(channelTitle && { title: channelTitle }),
      link: `${domain}`,
      ...(channelDescription && { description: channelDescription }),
      lastBuildDate: moment
        .utc(new Date())
        .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
      ...(feedLanguage && { language: feedLanguage }),
      ...(channelCategory && { category: channelCategory }),
      ...(channelCopyright && {
        copyright: channelCopyright,
      }), // TODO Add default logic
      ...(channelTTL && { ttl: channelTTL }),

      ...(channelLogo && {
        image: {
          url: buildResizerURL(channelLogo, resizerKey, resizerURL),
          title: channelTitle || feedTitle,
          link: domain,
        },
      }),

      item: elements.map((s) => {
        let body, category,
        enclosureurl = jmespath.search(s, audioAvailable);
        const url = `${domain}${s.website_url || s.canonical_url || ''}`
        return {
          title: jmespath.search(s, itemTitle) || '',
          link: url,
          guid: s._id,
          pubDate: s[pubDate],
          ...(enclosureurl  &&
          {
            enclosure: {
              '@url': enclosureurl,
              '@label':  getURLType(enclosureurl).label,
              '@group': getURLType(enclosureurl).label,
              '@type': getURLType(enclosureurl).type
            }
          }),
          ...(itemCategory &&
            (category = jmespath.search(s, itemCategory)) &&
            category && { category: category }),
            description: cheerio
            .load(
              rssBuildContent.parse(
                s.content_elements || [],
                includeContent,
                domain,
                resizerKey,
                resizerURL,
                resizerWidth,
                resizerHeight,
              ) || '',
            )
            .text()
        }
      }),
    },
  },
})

export function Rss({ globalContent, customFields, arcSite }) {
  const {
    resizerURL = '',
    feedDomainURL = '',
    feedTitle = '',
    feedLanguage = '',
  } = getProperties(arcSite)

  const { width = 0, height = 0 } = customFields.resizerKVP || {};
  

  const rssBuildContent = new BuildContent()
 
  // can't return null for xml return type, must return valid xml template
  return rssTemplate(get(globalContent, 'content_elements', []), {
    ...customFields,
    resizerURL,
    resizerWidth: width,
    resizerHeight: height,
    domain: feedDomainURL,
    feedTitle,
    feedLanguage,
    rssBuildContent,
  })
}

Rss.propTypes = {
  customFields: PropTypes.shape({
    audioAvailable: PropTypes.string.tag({
      label: 'Audio',
      group: 'Audio',
      description: 'description',
      defaultValue: `content_elements[?type=='audio'].streams[].url|[0]`,
    }),
    ...generatePropsForFeed('rss', PropTypes, ['featuredImage', 'includePromo']),
  }),
}
Rss.label = 'RSS Alexa'
export default Consumer(Rss)
