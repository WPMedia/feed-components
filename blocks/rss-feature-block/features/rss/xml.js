import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import get from 'lodash/get'
import moment from 'moment'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { BuildContent } from '@wpmedia/feeds-content-elements'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'
import { buildResizerURL } from '@wpmedia/feeds-resizer'

const jmespath = require('jmespath')

const rssTemplate = (
  elements,
  {
    channelTitle,
    channelDescription,
    channelPath,
    channelCopyright,
    channelTTL,
    channelUpdatePeriod,
    channelUpdateFrequency,
    channelCategory,
    channelLogo,
    imageTitle,
    imageCaption,
    imageCredits,
    itemTitle,
    itemDescription,
    pubDate,
    itemCategory,
    includePromo,
    includeContent,
    resizerURL,
    domain,
    feedTitle,
    feedLanguage,
    rssBuildContent,
  },
) => ({
  rss: {
    '@xmlns:atom': 'http://www.w3.org/2005/Atom',
    '@xmlns:content': 'http://purl.org/rss/1.0/modules/content/',
    '@xmlns:dc': 'http://purl.org/dc/elements/1.1/',
    '@xmlns:sy': 'http://purl.org/rss/1.0/modules/syndication/',
    '@version': '2.0',
    ...(includePromo && {
      '@xmlns:media': 'http://search.yahoo.com/mrss/',
    }),
    channel: {
      title: `${channelTitle || feedTitle}`,
      link: `${domain}`,
      'atom:link': {
        '@href': `${domain}${channelPath}`,
        '@rel': 'self',
        '@type': 'application/rss+xml',
      },
      description: `${channelDescription || feedTitle + ' News Feed'}`,
      lastBuildDate: moment
        .utc(new Date())
        .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
      ...(feedLanguage && { language: feedLanguage }),
      ...(channelCategory && { category: channelCategory }),
      ...(channelCopyright && {
        copyright: channelCopyright,
      }), // TODO Add default logic
      ...(channelTTL && { ttl: channelTTL }),
      ...(channelUpdatePeriod && {
        'sy:updatePeriod': channelUpdatePeriod,
      }),
      ...(channelUpdateFrequency && {
        'sy:updateFrequency': channelUpdateFrequency,
      }),
      ...(channelLogo && {
        image: {
          url: buildResizerURL(channelLogo, resizerKey, resizerURL),
          title: `${channelTitle || feedTitle}`,
          link: `${domain}`,
        },
      }),

      item: elements.map((s) => {
        let author, body, category
        const url = `${domain}${s.website_url || s.canonical_url}`
        const img =
          s.promo_items && (s.promo_items.basic || s.promo_items.lead_art)
        return {
          title: `${jmespath.search(s, itemTitle)}`,
          link: url,
          guid: {
            '#': url,
            '@isPermaLink': true,
          },
          ...((author = jmespath.search(s, 'credits.by[].name')) &&
            author && {
              'dc:creator': author.join(','),
            }),
          description: { $: jmespath.search(s, itemDescription) },
          pubDate: moment
            .utc(s[pubDate])
            .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
          ...(itemCategory &&
            (category = jmespath.search(s, itemCategory)) &&
            category && { category: category }),
          ...(includeContent !== '0' &&
            (body = rssBuildContent.parse(
              s.content_elements,
              includeContent,
              domain,
              resizerKey,
              resizerURL,
            )) &&
            body && {
              'content:encoded': {
                $: body,
              },
            }),
          ...(includePromo &&
            img &&
            img.url && {
              'media:content': {
                '@type': 'image/jpeg',
                '@url': buildResizerURL(img.url, resizerKey, resizerURL),
                ...(jmespath.search(img, imageCaption) && {
                  'media:description': {
                    '@type': 'plain',
                    $: jmespath.search(img, imageCaption),
                  },
                }),
                ...(jmespath.search(img, imageTitle) && {
                  'media:title': {
                    $: jmespath.search(img, imageTitle),
                  },
                }),
                ...((jmespath.search(img, imageCredits) || []).length && {
                  'media:credit': {
                    '@role': 'author',
                    '@scheme': 'urn:ebu',
                    '#': jmespath.search(img, imageCredits).join(','),
                  },
                }),
              },
            }),
        }
      }),
    },
  },
})

export function Rss({ globalContent, customFields, arcSite }) {
  const {
    resizerURL = '',
    feedDomainURL = '',
    feedTitle = '',
    feedLanguage = '',
  } = getProperties(arcSite)

  const rssBuildContent = new BuildContent()

  // can't return null for xml return type, must return valid xml template
  return rssTemplate(get(globalContent, 'content_elements', []), {
    ...customFields,
    resizerURL,
    domain: feedDomainURL,
    feedTitle,
    feedLanguage,
    rssBuildContent,
  })
}

Rss.propTypes = {
  customFields: PropTypes.shape({
    channelPath: PropTypes.string.tag({
      label: 'Path',
      group: 'Channel',
      description:
        'Path to the feed, excluding the domain, defaults to /arcio/rss',
      defaultValue: '/arcio/rss/',
    }),
    ...generatePropsForFeed('rss', PropTypes),
  }),
}
Rss.label = 'Standard RSS'
export default Consumer(Rss)
