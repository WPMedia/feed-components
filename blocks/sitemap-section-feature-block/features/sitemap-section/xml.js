import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import getProperties from 'fusion:properties'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'

const sitemapSectionTemplate = ({
  feedPath,
  feedParam,
  excludeSections,
  domain,
  sections,
  priority,
  changeFreq,
  buildIndexes,
}) => ({
  urlset: {
    '@xmlns': 'http://www.sitemaps.org/schemas/sitemap/0.9',
    url: buildIndexes(
      sections,
      excludeSections,
      domain,
      feedPath,
      feedParam,
      priority,
      changeFreq,
    ),
  },
})

export function SitemapSection({ globalContent, customFields, arcSite }) {
  const { feedDomainURL = '' } = getProperties(arcSite)
  const { children: sections } = globalContent
  const { excludeLinks = true } = customFields

  const checkForLinks = (section) => {
    if (excludeLinks && section.node_type === 'link') return true
  }

  const buildSitemapSectionLinks = (
    sections,
    excludeSections,
    domain,
    feedPath,
    feedParam,
    priority,
    changeFreq,
  ) => {
    const parameters = feedParam ? `?${feedParam}` : ''
    return sections.reduce((accum, section) => {
      const sectionId = section._id || ''
      if (
        sectionId &&
        !excludeSections.includes(sectionId) &&
        !checkForLinks(section)
      ) {
        accum.push({
          loc: `${domain}${feedPath ?? ''}${sectionId}/${parameters}`,
          ...(changeFreq !== 'Exclude field' &&
            changeFreq !== 'Exclude from sitemap' && {
              changefreq: changeFreq,
            }),
          ...(priority !== 'Exclude field' &&
            changeFreq !== 'Exclude from sitemap' && { priority: priority }),
        })
        if (section.children?.length) {
          const subSections = buildSitemapSectionLinks(
            section.children,
            excludeSections,
            domain,
            feedPath,
            feedParam,
            priority,
            changeFreq,
          )
          accum.push(...subSections)
        }
      }
      return accum
    }, [])
  }

  // can't return null for xml return type, must return valid xml template
  return sitemapSectionTemplate({
    ...customFields,
    excludeSections: (customFields.excludeSections || '').split(','),
    domain: feedDomainURL,
    sections,
    buildIndexes: buildSitemapSectionLinks,
  })
}

SitemapSection.propTypes = {
  customFields: PropTypes.shape({
    feedPath: PropTypes.string.tag({
      label: 'Feed Path',
      group: 'Format',
      description: 'Relative URL path used in the generated link',
      defaultValue: '',
    }),
    feedParam: PropTypes.string.tag({
      label: 'URL Parameters',
      group: 'Format',
      description:
        'Optional parameters to append to URL, join multiple parameters with &',
      defaultValue: '',
    }),
    excludeSections: PropTypes.string.tag({
      label: 'Section IDs to exclude',
      group: 'Format',
      description:
        'Comma separated list of section IDs to exclude from the feed, i.e /subscribe,/test',
      defaultValue: '',
    }),
    excludeLinks: PropTypes.bool.tag({
      label: 'Exclude links',
      group: 'Format',
      description: 'Should links be excluded from the output',
      defaultValue: true,
    }),
    ...generatePropsForFeed('sitemap', PropTypes, [
      'includePromo',
      'promoItemsJmespath',
      'resizerKVP',
      'lastMod',
    ]),
  }),
}
SitemapSection.label = 'Sitemap Section'
SitemapSection.icon = 'browser-page-hierarchy'
export default Consumer(SitemapSection)
