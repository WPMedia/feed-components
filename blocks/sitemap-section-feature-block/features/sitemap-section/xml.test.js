// eslint-disable-next-line no-unused-vars
import Consumer from 'fusion:consumer'
import { SitemapSection } from './xml'

const siteService = {
  children: [
    { _id: '/politics' },
    { _id: '/opinion' },
    { _id: '/economy' },
    {
      _id: '/sports',
      children: [
        { _id: '/sports/football' },
        { _id: '/sports/baseball' },
        { _id: '/sports/basketball' },
        { _id: 'link-CYAF5RAZGN023DTGKNWW4GWRMG', node_type: 'link' },
      ],
    },
    {
      _id: '/entertainment',
      children: [
        { _id: '/entertainment/tv' },
        { _id: '/entertainment/movies' },
      ],
    },
  ],
}

it('returns template with sitemap by section links', () => {
  const sitemapsection = SitemapSection({
    arcSite: 'demo',
    globalContent: siteService,
    customFields: {
      feedPath: '/arc/outboundfeeds/sitemap/category',
      feedParam: '',
      priority: '0.5',
      changeFreq: 'always',
    },
  })
  expect(sitemapsection).toMatchSnapshot()
})

it('returns template with sitemap by section links with excluded links', () => {
  const sitemapsection = SitemapSection({
    arcSite: 'demo',
    globalContent: siteService,
    customFields: {
      feedPath: '/arc/outboundfeeds/sitemap/category',
      feedParam: 'outputType=xml',
      excludeSections: '/politics,/entertainment/tv',
      priority: '0.5',
      changeFreq: 'always',
    },
  })
  expect(sitemapsection).toMatchSnapshot()
})

it('returns template with links to sections without outputType', () => {
  const sitemapsection = SitemapSection({
    arcSite: 'demo',
    globalContent: siteService,
    customFields: {
      feedPath: '',
      feedParam: '',
      excludeSections: '',
      priority: '',
      changeFreq: '',
    },
  })
  expect(sitemapsection).toMatchSnapshot()
})

it('returns template without priority or change frequency', () => {
  const sitemapsection = SitemapSection({
    arcSite: 'demo',
    globalContent: siteService,
    customFields: {
      feedPath: '/arc/outboundfeeds/sitemap/category',
      feedParam: 'outputType=xml',
      excludeSections: '',
      priority: 'Exclude field',
      changeFreq: 'Exclude field',
    },
  })
  expect(sitemapsection).toMatchSnapshot()
})
