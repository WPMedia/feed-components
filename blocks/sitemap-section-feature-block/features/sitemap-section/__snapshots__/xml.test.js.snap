// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns template with links to sections without outputType 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "url": Array [
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/politics/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/opinion/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/economy/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/sports/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/sports/football/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/sports/baseball/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/sports/basketball/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/entertainment/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/entertainment/tv/",
        "priority": "",
      },
      Object {
        "changefreq": "",
        "loc": "http://demo-prod.origin.arcpublishing.com/entertainment/movies/",
        "priority": "",
      },
    ],
  },
}
`;

exports[`returns template with sitemap by section links 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "url": Array [
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/politics/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/opinion/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/economy/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/sports/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/sports/football/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/sports/baseball/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/sports/basketball/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/entertainment/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/entertainment/tv/",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/entertainment/movies/",
        "priority": "0.5",
      },
    ],
  },
}
`;

exports[`returns template with sitemap by section links with excluded links 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "url": Array [
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/opinion/?outputType=xml",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/economy/?outputType=xml",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/sports/?outputType=xml",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/sports/football/?outputType=xml",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/sports/baseball/?outputType=xml",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/sports/basketball/?outputType=xml",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/entertainment/?outputType=xml",
        "priority": "0.5",
      },
      Object {
        "changefreq": "always",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/entertainment/movies/?outputType=xml",
        "priority": "0.5",
      },
    ],
  },
}
`;

exports[`returns template without priority or change frequency 1`] = `
Object {
  "urlset": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "url": Array [
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/politics/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/opinion/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/economy/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/sports/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/sports/football/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/sports/baseball/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/sports/basketball/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/entertainment/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/entertainment/tv/?outputType=xml",
      },
      Object {
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap/category/entertainment/movies/?outputType=xml",
      },
    ],
  },
}
`;
